File: home/src/main/java/com/changanford/home/search/adapter/SearchHistoryAdapter.java
package com.changanford.home.search.adapter;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.Nullable;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.changanford.home.R;
import com.changanford.home.room.SearchRecordEntity;
import java.util.List;
public class SearchHistoryAdapter extends BaseQuickAdapter<SearchRecordEntity, BaseViewHolder> {

    public boolean isExpand = false;

    public SearchHistoryAdapter(@Nullable List<SearchRecordEntity> data) {
        super(R.layout.item_common_search_tag, data);
    }

    public void setExpand(boolean isExpand) {
        this.isExpand = isExpand;
        notifyDataSetChanged();
    }

    @Override
    protected void convert(BaseViewHolder helper, SearchRecordEntity item) {
        TextView tvLabel = helper.getView(R.id.tv_search_tag);
        ImageView tvMore = helper.getView(R.id.iv_more_down);
        if (helper.getAdapterPosition() == 7 && !isExpand) {
            tvMore.setVisibility(View.VISIBLE);
            tvLabel.setText("");
        } else {
            tvMore.setVisibility(View.GONE);
            tvLabel.setText(item.getKeyword());
        }
    }
}

File: home/src/main/java/com/changanford/home/search/adapter/SearchHistoryAdapter.kt
package com.changanford.home.search.adapter

import androidx.lifecycle.MutableLiveData
import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseDataBindingHolder
import com.changanford.common.util.room.SearchRecordEntity
import com.changanford.home.R
import com.changanford.home.databinding.ItemCommonSearchTagBinding

class SearchHistoryAdapter :
    BaseQuickAdapter<SearchRecordEntity, BaseDataBindingHolder<ItemCommonSearchTagBinding>>(R.layout.item_common_search_tag) {

    var isExpand = MutableLiveData(false)
    var isShowEnd = false

    override fun convert(
        holder: BaseDataBindingHolder<ItemCommonSearchTagBinding>,
        item: SearchRecordEntity
    ) {
        holder.dataBinding?.let {
            it.tvSearchTag.text = item.keyword
        }
    }

    override fun getItemCount(): Int {
        if (data.size > 20) return 20
        return super.getItemCount()
    }
}
